### PR TITLE
TINY-9193: The caret is moved to the previous line after Enter key pressed

### PR DESCRIPTION
**Pull request by vpyshnenko**
Related Ticket: TINY-9193

Description of Changes:
* Placeholder text

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] ~Docs ticket created (if applicable)~

GitHub issues (if applicable):

### CODE REVIEW

**Comment by vpyshnenko** [comment]
Something wrong on CI:
```
Unexpected Bedrock Error -> Server Quitting **********

[2022-11-03T15:27:38.752Z] Unexpected runner error: Error

[2022-11-03T15:27:38.752Z] 

[2022-11-03T15:27:38.752Z] Stack:

[2022-11-03T15:27:38.752Z] promiseReactionJob@[native code]

[2022-11-03T15:27:45.495Z] Warning: Task "bedrock-auto:browser" failed. Use --force to continue.
```

File: modules/tinymce/src/core/test/ts/browser/textpatterns/TextPatternsTest.ts [hunk 1]
```diff
@@ -303,4 +304,24 @@ describe('browser.tinymce.core.textpatterns.TextPatternsTest', () => {
...
     editor.options.unset('text_patterns');
   });
+
+  it('TINY-9193: should move the caret to the next line after the Enter key pressed', () => {
+    const editor = hook.editor();
+    editor.options.set('text_patterns', [
+      { start: '~', end: '~', cmd: 'mceInsertContent', value: 'world' }
+    ]);
+    editor.setContent('<p>Hello</p><p>~test~</p>');
+    TinySelections.setCursor(editor, [ 1, 0 ], 6 );
```
Comment (lorenzo-pomili) [inline modules/tinymce/src/core/test/ts/browser/textpatterns/TextPatternsTest.ts:314]:
```suggestion
    editor.setContent('<p>Hello<br /></p><p>~test</p>');
    TinySelections.setCursor(editor, [ 1, 0 ], 5 );
    TinyContentActions.type(editor, '~');
```
I tried to run the test with the old code but it seems to not make the test fail.
With this close seems to fail without the fix and work with it but I'm not sure that is it the correct way to do it

**Review by lorenzo-pomili — CHANGES_REQUESTED** [review]

File: modules/tinymce/src/core/test/ts/browser/textpatterns/TextPatternsTest.ts (hunk 1 above)
Comment (vpyshnenko) [reply modules/tinymce/src/core/test/ts/browser/textpatterns/TextPatternsTest.ts:314]: Did you try the test with the old code in Firefox/MacOS?

**Review by TheSpyder — DISMISSED** [review]
Sorry, you can't just remove the "delete" line from the "delete selected content" function. Before even getting to why it's there, now the function only does what it says some of the time.

There's a very good reason why we delete selected content before inserting and I would want to see far more evidence and discussion about why this is the best solution before I can consider accepting such a big change.

File: modules/tinymce/src/core/test/ts/browser/textpatterns/TextPatternsTest.ts (hunk 1 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/core/test/ts/browser/textpatterns/TextPatternsTest.ts:314]: actually in Firefox/Linux, but it seems to work as in Mac, now that you made me think, it is possible that in Linux it works differently

**Comment by vpyshnenko** [comment]
I've traced the issue down to the point of the deletion of the matched content through the native `Delete` command. Unfortunately in Firefox, this command implementation is accompanied by moving the caret to the previous block. While searching for a workaround I found that down the road the `insertHtmlAtCaret `passes through the `editor.selection.SetContent()` which internally ends up to `rng.deleteContents()`. Surprisingly removing the line with the `editor.getDoc().execCommand('Delete', false) `call passes all the tests. That makes me assume that it is a redundant call.  It that's not the case, probably we need to figure out a test case where we strictly depend on the `Delete `command for the deletion of ranged selection.

**Comment by TheSpyder** [comment]
> I found that down the road the `insertHtmlAtCaret `passes through the `editor.selection.SetContent()` which internally ends up to `rng.deleteContents()`.

Ah that's not as bad as I thought. Thanks for narrowing the condition, I'll look at it soon.

File: modules/tinymce/CHANGELOG.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Dragging a noneditable element before or after another noneditable element now works correctly. #TINY-9253
 - The restored selection after a redo or undo action was not scrolled into view. #TINY-9222
 - A newline could not be inserted when the selection was restored from a bookmark after an inline `contenteditable="false"` element. #TINY-9194
+- The caret was moved to the previous line when the inline text pattern have been executed on Enter key pressing in Firefox. #TINY-9193
```
Comment (spocke) [inline modules/tinymce/CHANGELOG.md:45]: Not sure if this is better:

```suggestion
- The caret was moved to the previous line when a text pattern executed a `mceInsertContent` command on Enter key when running on Firefox. #TINY-9193
```

**Review by spocke — APPROVED** [review]

**Review by lorenzo-pomili — APPROVED** [review]

**Pull request merged**